Frameworks/CoreGraphics/DWriteFontBinaryDataLoader.mm

```diff
@@ -0,0 +1,119 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "DWriteFontBinaryDataLoader.h"
+
+using namespace Microsoft::WRL;
+
+/**
+ * Private class.
+ * Custom implementation of IDWriteFontFileStream that implements Read in terms of an underlying CFDataRef
+ *
+ * While IDWriteFontFileStream normally frees its underlying data bit by bit through ReleaseFileFragment(),
+ * for WinObjC purposes, it is easier to rely on existing mechanisms for CFData's destruction.
+ * Thus, this class releases its CFData all at once, at the time of its destruction.
+ */
+class DWriteFontBinaryDataStream : public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IDWriteFontFileStream> {
+protected:
+    InspectableClass(L"Windows.Bridge.DirectWrite", TrustLevel::BaseTrust);
+
+public:
+    DWriteFontBinaryDataStream() {
+    }
+
+    HRESULT RuntimeClassInitialize(CFDataRef data) {
```

Comment:
yeah, but since this is only used from the below class, it's sufficient that we validate below.

---
In reply to: [90736864](https://github.com/Microsoft/WinObjC/pull/1504#discussion_r90736864) [](ancestors = 90736864)